data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -67,61 +71,69 @@ public PublishedNewsDisplayedPropUpgrade(InitParams initParams,
     this.container = container;
   }
 
-  public int getPublishedNewsDisplayedPropCount() {
-    return publishedNewsDisplayedPropCount;
+  public int getMigratedPublishedNewsCount() {
+    return migratedPublishedNewsCount;
   }
 
   @Override
   public void processUpgrade(String oldVersion, String newVersion) {
     long startupTime = System.currentTimeMillis();
     LOG.info("Start upgrade of published news displayed property");
-
-    boolean transactionStarted = false;
-    RequestLifeCycle.begin(container);
+    ExoContainerContext.setCurrentContainer(container);
```

Comment:
please add the above code inside `try {} finally { ExoContainerContext.setCurrentContainer(null); }` to avoid memory leak in the `ThreadLocal`